front/src/containers/modal-change-password/index.js

```diff
@@ -0,0 +1,127 @@
+                                    <Input.Password placeholder="Old password" />
+                                </Form.Item>}
+                                <Form.Item name="newPassword" align="center" rules={[{ required: true, message: 'Please input your new password!' }, { validator: validatePassword }]} >
+                                    <Input.Password placeholder={user === 'admin' ? 'Password' : 'New password'} />
+                                </Form.Item>
+                                <Form.Item name="confirmNewPassword" dependencies={['newPassword']} align="center"
+                                    rules={[
+                                        { required: true, message: 'Please input your new password again!' },
+                                        ({ getFieldValue }) => ({
+                                            validator(rule, value) {
+                                                if (!value || getFieldValue('newPassword') === value) {
+                                                    return Promise.resolve();
+                                                }
+                                                return Promise.reject('The two passwords that you entered do not match!');
+                                            },
+                                        }),
+                                        { validator: validatePassword }]} >
+                                    <Input.Password placeholder={user === 'admin' ? 'Confirm password' : 'Confirm new password'} />
+                                </Form.Item>
+                                <Form.Item name={user} style={{ display: 'none' }}>
+                                    <Input type="text" />
+                                </Form.Item>
+                                {this.props.teacherId ?
+                                    <Form.Item name={['teacherId', this.props.teacherId]} style={{ display: 'none' }}>
+                                        <Input type="text" />
+                                    </Form.Item>
+                                    : null}
+                                {errorMessage ? <ErrorText>{errorMessage}</ErrorText> : null}
+                                <Form.Item align="center" {...tailLayout}>
+                                    <Button type="primary" htmlType="submit"> {user === 'admin' ? 'Update password' : 'Change password'} </Button>
+                                </Form.Item>
+                            </Form>
+                        </Col>
+                    </Row>
+                </Modal></div>
+        )
+    }
+}
+
+const mapStateToProps = ({ changePassword: { errorMessage, isChangedPassword } }) => ({
+    errorMessage,
+    isChangedPassword
+});
+
+const mapDispatchToProps = (dispatch) => {
```

Comment:
Just use the plain object for `mapDispatchToProps`

Reply:
done